### PR TITLE
Add Filter Transaction Command 

### LINKED ISSUE #161: Implement filter by buy/sell for transactions
Users should be able to run a filter command and view only either buy or sell in the transactions

### PR DESCRIPTION
**Pull request by jicsontoh**
Implemented the Filter Transaction Command which filters all the transaction by buy or sell.

Updated the UI to reflect all the transactions based on the filter

Updated the Filter Command section in the UG

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/FilterTransCmdParser.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.address.logic.commands.FilterTransCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new FilterTransCommand object
```
Comment (notle1706) [inline src/main/java/seedu/address/logic/parser/FilterTransCmdParser.java]: I think you might be missing a fullstop here.

**Review by notle1706 — COMMENTED** [review]
You might want to check the javadocs

**Comment by jicsontoh** [comment]
CodeCov failed to cover due to changes in UI

File: src/main/java/seedu/address/logic/parser/FilterTransCmdParser.java (hunk 1 above)
Comment (jicsontoh) [reply src/main/java/seedu/address/logic/parser/FilterTransCmdParser.java]: Sure, thank you.

**Review by ElginL — APPROVED** [review]
LGTM! Good work

File: src/main/java/seedu/address/logic/commands/FilterTransCommand.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_CLIENTS;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+
+/**
+ * Filters the all the transaction by all the clients.
```
Comment (notle1706) [inline src/main/java/seedu/address/logic/commands/FilterTransCommand.java]: Perhaps this sentence should be 'Filters all the transaction by all the clients.` instead?

File: src/main/java/seedu/address/logic/parser/FilterTransCmdParser.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new FilterTransCommand object.
+ */
+public class FilterTransCmdParser implements Parser<FilterTransCommand> {
+    /**
+     * Parses {@code userInput} into a command and returns it.
+     *
+     * @param userInput
```
Comment (notle1706) [inline src/main/java/seedu/address/logic/parser/FilterTransCmdParser.java:15]: Should there be a fullstop after this sentence?

**Review by notle1706 — CHANGES_REQUESTED** [review]
Good coding, however there are some documentations that should be change.

File: src/main/java/seedu/address/logic/commands/FilterTransCommand.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    public FilterTransCommand(boolean isBuy) {
+        this.isBuy = isBuy;
+    }
+
+    /**
+     * Executes the command and returns the result message.
+     *
+     * @param model {@code Model} which the command should operate on.
+     * @return feedback message of the operation result for display
```
Comment (notle1706) [inline src/main/java/seedu/address/logic/commands/FilterTransCommand.java]: Perhaps this sentence should end with a fullstop.

**Review by notle1706 — CHANGES_REQUESTED** [review]
Code looks good but there are still some errors with the javdocs.

**Review by notle1706 — APPROVED** [review]
LGTM good job!

**Pull request merged**